lib/active_record/virtual_attributes/virtual_total.rb

```diff
@@ -72,21 +72,22 @@ def virtual_aggregate(name, relation, method_name = :sum, column = nil, options
       def define_virtual_aggregate_method(name, relation, method_name, column)
         if method_name == :size
           define_method(name) do
-            (attribute_present?(name) ? self[name] : nil) || send(relation).try(:size) || 0
+            has_attribute?(name) ? self[name] : (send(relation).try(:size) || 0)
           end
         else
           define_method(name) do
-            (attribute_present?(name) ? self[name] : nil) ||
-              begin
-                rel = send(relation)
-                if rel.loaded?
-                  rel.blank? ? nil : (rel.map { |t| t.send(column).to_i } || 0).send(method_name)
-                else
-                  # aggregates are not smart enough to handle virtual attributes
-                  arel_column = rel.klass.arel_attribute(column)
-                  rel.try(method_name, arel_column) || 0
-                end
+            if has_attribute?(name)
```

Comment:
I ended up pushing the branch I was working for the above comment for reference:

https://github.com/ManageIQ/activerecord-virtual_attributes/compare/master...NickLaMuro:fix_kbrock_commit_nightmare

Don't expect you to use this (too much snark/shade in it for one thing...), but it was much easier for me to make the changes this way instead of trying to pseudo code this.

Reply:
I tried to get close to your commits.

github is displaying them out of order. it should be:

```
cb353c754a - virtual_aggregate remove unnecessary [] || 0
1657a3be06 - better aggregate support
585d64ad4d - virtual_total nil attribute does not exec query
be6861b9a2 - virtual_aggregate use if else
09c9f31d8b - virtual_aggregate nil attribute does not exec query
216390386f - virtual_total: remove blank case
b1862d919e - For sum, coalesce sql values to 0
```